server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -508,7 +522,8 @@ private ClusterState applyCreateIndexRequestWithV1Templates(
         final CreateIndexClusterStateUpdateRequest request,
         final boolean silent,
         final List<IndexTemplateMetadata> templates,
-        final BiConsumer<Metadata.Builder, IndexMetadata> metadataTransformer
+        final BiConsumer<Metadata.Builder, IndexMetadata> metadataTransformer,
+        final ActionListener<Void> listener
```

Comment:
Could we call this `rerouteListener` (here and elsewhere in this PR) just to clarify what it's listening for?